core/primitives/src/types.rs

```diff
@@ -18,15 +18,29 @@ pub type StorageUsage = u64;
 pub type StorageUsageChange = i64;
 /// Nonce for transactions.
 pub type Nonce = u64;
-/// Index of the block.
-pub type BlockIndex = u64;
+/// Height of the block.
+pub type BlockHeight = u64;
 /// Shard index, from 0 to NUM_SHARDS - 1.
 pub type ShardId = u64;
 /// Balance is type for storing amounts of tokens.
 pub type Balance = u128;
 /// Gas is a type for storing amount of gas.
 pub type Gas = u64;
 
+/// Number of blocks in current group.
+pub type NumBlocks = u64;
+/// Number of shards in current group.
+pub type NumShards = u64;
+/// Number of seats of validators (block producer or hidden ones) in current group.
+pub type NumSeats = u64;
+/// Height delta that measures the difference between `BlockHeight`s.
+pub type HeightDelta = u64;
+
+#[derive(Default, BorshSerialize, BorshDeserialize, Clone, Copy, Debug, PartialEq, Eq)]
+pub struct Seat {
```

Comment:
It doesn't make sense to have this type for two reasons:
* Where this type is used it actually means an index in an array. I don't see why you want to wrap this type and makes code harder to read.
* Even if you want to wrap it, it is more idiomatic to have `pub struct Seat(ValidatorId)` so that the compiler will for sure optimizes away the wrapper.

Reply:
It's related to https://github.com/nearprotocol/nearcore/issues/1855. Let's discuss offline.

Reply:
Simplified to `pub type Seat = ValidatorId;`

Reply:
Removed `Seat` for now. We'll reopen this later if necessary.